### PR TITLE
BOAC-3964: API for creating a unit requirement

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-3964

### CODE REVIEW

File: boac/api/degree_progress_controller.py [hunk 1]
```diff
@@ -46,3 +47,25 @@ def create_degree():
         advisor_dept_codes=dept_codes_where_advising(current_user),
     )
     return tolerant_jsonify(degree.to_api_json())
+
+
+@app.route('/api/degree_progress/<template_id>/unit_requirement', methods=['POST'])
```
Comment (johncrossman) [inline boac/api/degree_progress_controller.py:52]: @lyttam  - What do you think about `/api/degree/...` as the API prefix?  That would match the other, and it's concise.

**Comment by johncrossman** [comment]
Makes sense. Posted one question.

File: boac/api/degree_progress_controller.py (hunk 1 above)
Comment (lyttam) [reply boac/api/degree_progress_controller.py:52]: @johncrossman I don't have strong feelings either way. `/api/degree_progress/...` matches the controller name, but concise is nice. Should we change the controller name to match, i.e. `degree_controller.py`?

Comment (johncrossman) [reply boac/api/degree_progress_controller.py:52]: IMO, the controller name can remain as is. Thanks @lyttam

**Pull request merged**